--
-- PostgreSQL database dump
--

-- Dumped from database version 9.5.1
-- Dumped by pg_dump version 9.5.1

SET statement_timeout = 0;
SET lock_timeout = 0;
SET client_encoding = 'UTF8';
SET standard_conforming_strings = on;
SET check_function_bodies = false;
SET client_min_messages = warning;
SET row_security = off;

SET search_path = public, pg_catalog;

--
-- Data for Name: cv; Type: TABLE DATA; Schema: public; Owner: -
--

COPY cv (cv_id, "group", term, definition, rank) FROM stdin;
1	project_prop	genotyping_purpose	The purpose of genotyping such as MABC, MARS, Diversity etc	1
2	project_prop	date_sampled	Date tissue sampled and approx date of field trial	2
3	project_prop	division	Department or division where the project was made	3
4	project_prop	study_name	The name of the research study	4
5	platform_type	GBS	Genotyping by Sequencing	0
6	platform_type	Illumina_Infinium	Illumina Infinium	0
7	platform_type	Illumina_Goldengate_bxp	Illumina GoldenGate	0
8	platform_type	KASP	KASP	0
9	platform_type	Sequencing	Sequence-based Technologies	0
10	platform_type	Affymetrix_Axiom	Affymetrix Axiom	0
11	platform_type	Dartseq_SNPs	DArTSeq SNP	0
12	platform_type	Dart_silico	Silico DArT	0
13	platform_type	Dart_clone	DArT Clone	0
14	platform_type	Fluidigm	Fluidigm	0
15	platform_type	SSR	Simple Sequence Repeats	0
16	platform_type	Indel	insertion-deletion variant	0
17	platform_type	SSR_STS_CAPS	Fragment-size polymorphism technologies including Simple Sequence Repeats, Sequence-Tagged-Sites, and Cleaved Amplified Polymorphic Sequences	0
18	germplasm_type	Inbred_line	Inbred line	1
19	germplasm_type	Population	Population	2
20	germplasm_type	F1_hybrid	F1_hybrid, the first filial generation or hybrid offspring in a genetic cross-fertilization, such as bi-parental, 3-way, 4-way breeding crosses or hybrid test cross. For GOBII, it represent mostly bi-parental F1 breeding cross.	3
21	germplasm_type	3_way_hybrid	3-way_F1_Cross, breeding cross involving 3 different parents	4
22	germplasm_type	4_way_hybrid	4-way_F1_Cross, breeding cross involving 4 different parents	5
23	germplasm_type	F2	F2, the second filial generation, offspring resulting from one cycle of selfing (S1) or interbreeding of the F1 breeding cross	6
24	germplasm_type	F3	F3, the third filial generation, offspring resulting from two cycles of selfing (S2) or interbreeding of the F1 breeding cross	7
25	germplasm_type	F4	F4, the fourth filial generation, offspring resulting from three cycles of selfing (S3) or interbreeding of the F1 breeding cross	8
26	germplasm_type	F5	F5, the fifth filial generation, offspring resulting from four cycles of selfing (S4) or interbreeding of the F1 breeding cross	9
27	germplasm_type	F6	F6, the sixth filial generation, offspring resulting from five cycles of selfing (S5) or interbreeding of the F1 breeding cross	10
28	germplasm_type	BC1F1	BC1F1, the first generation of F1 backcrossed to the recurrent parent with 75% recurrent parent recovery	11
29	germplasm_type	BC2F1	BC2F1, the second generation of F1 backcrossed to the recurrent parent with 87.5% recurrent parent recovery	12
30	germplasm_type	BC3F1	BC3F1, the third generation of F1 backcrossed to the recurrent parent with 93.75% recurrent parent recovery	13
31	germplasm_type	BC4F1	BC4F1, the fourth generation of F1 backcrossed to the recurrent parent with 96.875% recurrent parent recovery	14
32	germplasm_type	BC5F1	backcross 5; F1 generation	15
33	germplasm_type	BC1F2	backcross 1; F2 generation	16
34	germplasm_type	BC2F2	backcross 2; F2 generation	17
35	germplasm_type	BC3F2	backcross 3; F2 generation	18
36	germplasm_type	BC4F2	backcross 4; F2 generation	19
37	germplasm_type	BC5F2	backcross 5; F2 generation	20
38	germplasm_type	Landrace	A locally adapted regional ecotype that may also be called a traditional variety	21
39	germplasm_type	Synthetic	A variety generated by crossing in all combinations a series of lines	22
40	germplasm_type	RIL	RIL: Recombinant Inbred Population	23
41	germplasm_type	OPV	OPV: Open-pollinated variety	24
42	germplasm_type	NIL	NIL: Near Isogenic Lines	25
43	germplasm_type	DH	DH: Doubled Haploid	26
44	germplasm_type	NAM	NAM: Nested Association Mapping	27
45	germplasm_type	MAGIC	MAGIC: Multi Parent Advanced Generation Intercrossed Population	28
46	germplasm_prop	germplasm_id	This will be a higher level GID eg MGID describing a group of similar lines/genotypes	1
47	germplasm_prop	seed_source_id	Seed source ID	2
48	germplasm_prop	germplasm_subsp	Germplasm sub-species eg indica, japonica, for rice; Dent, Flint for maize	3
49	germplasm_prop	germplasm_heterotic_group	Heterotic groups within species eg NSS, SSS, A, B for maize	4
50	germplasm_prop	par1	Parent 1 of the germplasm name	5
51	germplasm_prop	par2	Parent 2 of the germplasm name	6
52	germplasm_prop	par3	Parent 3 of the germplasm	7
53	germplasm_prop	par4	Parent 4 of the germplasm name 	8
54	germplasm_prop	pedigree		9
55	status	new	new row	1
56	status	modified	modified row	2
57	status	deleted	deleted row	3
58	mapset_type	physical	Physical map	1
59	mapset_type	genetic	Genetic map	2
60	marker_strand	TOP	Marker design on the 'top' strand as defined by Illumina	1
61	marker_strand	BOT	Marker design on the 'bottom' strand as defined by Illumina	2
62	marker_strand	Forward	Marker designed on the Forward Strand - used by most technologies apart from Illumina and Affymetrix	3
63	marker_strand	+	Marker designed on the positive or + strand eg used by  Affymetrix	4
64	marker_prop	primer_forw1	First forward primer	1
65	marker_prop	primer_forw2	Second forward primer	2
66	marker_prop	primer_rev1	First reverse primer	3
67	marker_prop	primer_rev2	Second reverse primer	4
68	marker_prop	probe1	First probe	5
69	marker_prop	probe2	Second probe	6
70	marker_prop	polymorphism_type	This describes the type of polymorphism that the marker interrogates eg SSR;SNP;Indel.	7
71	marker_prop	synonym	Another name for the marker	8
72	marker_prop	source	The source of the marker eg a publication reference or sequence assembly	9
73	marker_prop	gene_id	The gene the marker was designed to	10
74	marker_prop	gene_annotation	Type of gene eg transcription factor	11
75	marker_prop	polymorphism_annotation	eg Synonymous etc	12
76	marker_prop	marker_dom	Describes if the marker behaves in a dominant fashion ie  presence/ absence with allele dosage (ie heterozygous or homozygous presence) for presence being unknown	13
77	marker_prop	clone_id	The clone ID - used for Dart technologies = the MarkerName for Silico Dart	14
78	marker_prop	clone_id_pos	The SNP position on the sequence for the clone ID - used for Dart technologies	15
79	marker_prop	genome_build	The genome build that the marker was made with	16
80	marker_prop	typeofrefallele_alleleorder	How a reference allele is called eg TOP/BOTT, F/R, +/-, first found, minor/major etc	17
81	marker_prop	strand_data_read	Strand that the allele data is read on ie the probe is designed to	18
82	marker_prop	allele2	second alternative allele	19
83	marker_prop	allele3	third alternative allele	20
84	dnasample_prop	trial_name	Trial name for field experiment that the sample is coming from, or fieldbook	1
85	dnasample_prop	sample_group	Sample Group eg MABCTHC_cycle1	2
86	dnasample_prop	sample_group_cycle	Cycle eg backcrossing cycle for a sample group	3
87	dnasample_prop	sample_type	Type of tissue sampled eg leaf, seed, bulk seed, bulk plant	4
88	dnasample_prop	sample_parent_prop	Type of parent for a population eg donor, recurrent etc	5
89	dnasample_prop	ref_sample	Standard sample against which all other germplasm is compared to for this GID/MGID. Gold standard or Reference line.	6
90	dnarun_prop	barcode	Sample DNA barcode used to identify a sample using sequence-based technologies eg ACAATGGA	1
91	germplasm_species	Zea_mays_subsp._mays		1
92	germplasm_species	Zea_mays_subsp._mexicana		2
93	germplasm_species	Zea_mays_subsp._parviglumis		3
94	germplasm_species	Tripsacum_dactyloides		4
95	germplasm_species	Unknown		5
96	germplasm_species	Triticum_aestivum_subsp_aestivum		6
97	germplasm_species	Triticum_turgidum_subsp_durum  		7
98	germplasm_species	Triticum_turgidum_subsp._polonicum		8
99	germplasm_species	Triticum_turgidum_subsp._turgidum		9
100	germplasm_species	Triticum_turgidum_subsp._carthlicum		10
101	germplasm_species	Triticum_turgidum_subsp._turanicum		11
102	germplasm_species	Triticum_aestivum_subsp._compactum		12
103	germplasm_species	Triticum_turgidum_subsp._dicoccon		13
104	germplasm_species	Triticum_aestivum_subsp._spelta		14
105	germplasm_species	Secale_cereale		15
106	germplasm_species	Aegilops_squarrosa		16
107	germplasm_species	Triticum_monococcum_subsp._monococcum		17
108	germplasm_species	Triticum_turgidum_subsp._dicoccoides		18
109	germplasm_species	Triticum_aestivum_subsp._sphaerococcum		19
110	germplasm_species	Triticum_urartu		20
111	germplasm_species	Triticum_boeoticum		21
112	germplasm_species	Triticum_monococcum		22
113	germplasm_species	Aegilops_speltoides		23
114	germplasm_species	Hordeum_vulgare		24
115	germplasm_species	Aegilops_biuncialis		25
116	germplasm_species	Aegilops_kotschyi		26
117	germplasm_species	Aegilops_longissima		27
118	germplasm_species	Aegilops_umbellulata		28
119	germplasm_species	Aegilops_cylindrica		29
120	germplasm_species	Aegilops_geniculata		30
121	germplasm_species	Aegilops_columnaris		31
122	germplasm_species	Aegilops_comosa		32
123	germplasm_species	Aegilops_peregrina		33
124	germplasm_species	Triticum_aestivum_subsp._macha		34
125	germplasm_species	Triticum_timopheevii_subsp._timopheevii		35
126	germplasm_species	Triticum_monococcum_subsp._aegilopoides		36
127	germplasm_species	Aegilops_triuncialis		37
128	germplasm_species	Aegilops_tauschii_subsp._strangulata		38
129	germplasm_species	Aegilops_tauschii_var._meyerii		39
130	germplasm_species	Hordeum_vulgare_subsp._vulgare		40
131	germplasm_species	Triticum_zhukovskyi		41
132	germplasm_species	Triticum_turgidum_subsp._paleocolchicum		42
133	germplasm_species	Triticum_timopheevii_subsp._armeniacum		43
134	germplasm_species	Triticum_spp.		44
135	germplasm_species	Aegilops_crassa		45
136	germplasm_species	Triticum_hybr.		46
137	germplasm_species	Triticum_vavilovii		47
138	germplasm_species	x_Tritordeum_spp.		48
139	germplasm_species	Aegilops_tauschii_x_Triticum_turgidum		49
140	germplasm_species	x_Aegilotriticum_spp.		50
141	germplasm_species	x_Triticosecale_spp.		51
142	germplasm_species	x_Triticoaegilops_spp.		52
143	germplasm_species	Undetermined_spp.		53
144	germplasm_species	Aegilops_tauschii		54
145	germplasm_species	Agropyron_spp.		55
146	germplasm_species	Oryza_sativa		56
147	germplasm_species	Oryza_australiensis		57
148	germplasm_species	Oryza_barthii		58
149	germplasm_species	Oryza_glaberrima		59
150	germplasm_species	Oryza_longistaminata		60
151	germplasm_species	Oryza_punctata		61
152	germplasm_species	Oryza_rufipogon		62
153	germplasm_species	Cicer_arietinum		63
154	germplasm_species	Cicer_reticulatum		64
155	germplasm_species	Cicer_echinospermum		65
156	germplasm_species	Sorghum_bicolor		66
157	analysis_type	calling	Sequence alignment and variant calling pipelines	1
158	analysis_type	cleaning	Data cleaning methods used to remove poor quality data	2
159	analysis_type	imputation	Imputation methods	3
160	analysis_type	allele_sorting	Method for sorting alleles eg due to phasing	4
161	dataset_type	nucleotide_2_letter	eg AA CC CT for SNPs, + - for indels and NN for missing. Any allele phasing will be maintained	1
162	dataset_type	iupac	IUPAC eg SNPs = ACGRY, indels = + 0 - , missing = N	2
163	dataset_type	dominant_non_nucleotide	0, 1 , N: 0 can be absence and 1 can be presence. N = missing	3
164	dataset_type	co_dominant_non_nucleotide	0, 1, 2, N: 0 can be absence, 1 is the heterozygote and 2 can be presence. N = missing	4
165	dataset_type	ssr_allele_size	Used for SSR allele sizes - converted to 8 numbers eg 123/125 becomes 01230125. Missing = 00000000	5
\.


--
-- Name: cv_cv_id_seq; Type: SEQUENCE SET; Schema: public; Owner: -
--

SELECT pg_catalog.setval('cv_cv_id_seq', 166, true);


--
-- PostgreSQL database dump complete
--

